test/test_images/latencymako/sender_receiver.go

```diff
@@ -358,121 +360,102 @@ func warmup(warmupSeconds uint) {
-
-	return cancel
+	return nil
 }
 
-func processVegetaResult(vegetaResults <-chan *vegeta.Result, doneCh chan<- struct{}) {
+// processVegetaResult processes the results from the Vegeta attackers.
+func (ex *senderReceiverExecutor) processVegetaResult(vegetaResults <-chan *vegeta.Result, doneCh chan<- struct{}) {
 	// Discard all vegeta results and wait the end of this channel
 	for _ = range vegetaResults {
 	}
 
 	printf("All requests sent")
 
-	close(sentCh)
-	close(acceptedCh)
+	close(ex.sentCh)
+	close(ex.acceptedCh)
+	close(ex.failedCh)
 
 	// assume all responses are received after a certain time
 	time.Sleep(8 * time.Second)
-	close(receivedCh)
+	close(ex.receivedCh)
 
 	printf("All channels closed")
 
 	close(doneCh)
 }
 
-func processReceiveEvent(event cloudevents.Event) {
-	receivedCh <- receivedState{eventId: event.ID(), at: ptypes.TimestampNow()}
+// processReceiveEvent processes the event received by the CloudEvents receiver.
+func (ex *senderReceiverExecutor) processReceiveEvent(event cloudevents.Event) {
+	ex.receivedCh <- receivedState{eventId: event.ID(), at: ptypes.TimestampNow()}
 }
 
 // processEvents keeps a record of all events (sent, accepted, failed, received).
-func processEvents() {
+func (ex *senderReceiverExecutor) processEvents() {
 	for {
 		select {
 
-		case e, ok := <-sentCh:
+		case e, ok := <-ex.sentCh:
 			if !ok {
 				continue
 			}
-			sentEvents.Events[e.eventId] = e.at
+			ex.sentEvents.Events[e.eventId] = e.at
 
-		case e, ok := <-acceptedCh:
+		case e, ok := <-ex.acceptedCh:
 			if !ok {
 				continue
 			}
 
-			_, sent := sentEvents.Events[e.eventId]
+			_, sent := ex.sentEvents.Events[e.eventId]
 			if !sent {
 				// Send timestamp not yet recorded, reenqueue
-				acceptedCh <- e
+				ex.acceptedCh <- e
 				continue
 			}
 
-			acceptedEvents.Events[e.eventId] = e.at
+			ex.acceptedEvents.Events[e.eventId] = e.at
 
-		case e, ok := <-failedCh:
+		case e, ok := <-ex.failedCh:
 			if !ok {
 				continue
 			}
 
-			_, sent := sentEvents.Events[e.eventId]
+			_, sent := ex.sentEvents.Events[e.eventId]
 			if !sent {
 				// Send timestamp not yet recorded, reenqueue
-				failedCh <- e
+				ex.failedCh <- e
 				continue
 			}
 
-			failedEvents.Events[e.eventId] = e.at
+			ex.failedEvents.Events[e.eventId] = e.at
 
-		case e, ok := <-receivedCh:
+		case e, ok := <-ex.receivedCh:
 			if !ok {
 				return
 			}
 
-			receivedEvents.Events[e.eventId] = e.at
+			ex.receivedEvents.Events[e.eventId] = e.at
 		}
 	}
 }
 
-func aggregatorClient() (cli pb.EventsRecorderClient, closeConnFn func() error) {
-	conn, err := grpc.Dial(aggregAddr, grpc.WithInsecure())
-	if err != nil {
-		fatalf("Failed to connect to the aggregator: %v", err)
-	}
-
-	return pb.NewEventsRecorderClient(conn), conn.Close
-}
-
-func sendEventsRecordList(cli pb.EventsRecorderClient, rl *pb.EventsRecordList) error {
+func (ex *senderReceiverExecutor) sendEventsRecordList(rl *pb.EventsRecordList) error {
```

Comment:
This method should also take a context as parameter instead of using background.